SeleniumTests/LoginTests.cs

```diff
@@ -49,84 +49,90 @@ public void CloseBrowser()
-            //frowned upon in general, but is often unavoidable)
+            // waits for error message to appear
             System.Threading.Thread.Sleep(200);
 
-            //** NOTE: Now see if the error message is displayed
+            // grabs error message
             var errorMessageIsDisplayed = _driver.FindElement(By.Id("errorMessage")).Displayed;
 
-            //** NOTE: The Assert class has static methods that will fail a test if the condition does not pass
+            // asserts it displayed correctly
             Assert.IsTrue(errorMessageIsDisplayed, "No 'Invalid Username or Password' message displayed");
 
-            //The error message should disappear
+            // waits for the disappear
             Thread.Sleep(5000);
+
+            // tries to grab the message
             errorMessageIsDisplayed = _driver.FindElement(By.Id("errorMessage")).Displayed;
+
+            // assers it was gone
             Assert.IsFalse(errorMessageIsDisplayed, "Error message is displayed but should have faded out");
 
-            
+
         }
 
         [TestMethod]
         public void AValidUserName_WithAnInvalidPassword()
         {
-            //** HINT: View the login page in the brower,  Valid username and password combinations have been provided right from the login page
-            //         Examine the test above and write the test where the username is valid, but the password is invalid
-            throw new NotImplementedException("Test not implemented");
-            
-            /*Assert:
-             * Error message displays and then fades out
-             */
+            do_login("admin", "bad_password");
+
+            System.Threading.Thread.Sleep(200);
+
+            var errorMessageIsDisplayed = _driver.FindElement(By.Id("errorMessage")).Displayed;
+
+            Assert.IsTrue(errorMessageIsDisplayed, "No 'Invalid Username or Password' message displayed");
+
+            Thread.Sleep(5000);
+
+            errorMessageIsDisplayed = _driver.FindElement(By.Id("errorMessage")).Displayed;
+            Assert.IsFalse(errorMessageIsDisplayed, "Error message is displayed but should have faded out");
         }
 
         [TestMethod]
         public void A_Physician_Logs_In_Successfully()
         {
-            throw new NotImplementedException("Test not implemented");
-           
-            /*Assert:
-             * The physician is taken to the physician page (iemosoft.com/selenium-test-physician)
-             * HINT: use _driver.Url
-             */
+            // does doc login
+            do_login("dr smith", "P@ssword");
+
+
+            // waits for page to load
+            Thread.Sleep(3000);
+
+            // grabs url
+            String doc_url = _driver.Url;
+
+            // compares expected url to actual and assters they are equil
+            Assert.AreEqual("http://iemosoft.com/selenium-test-physician/", doc_url);
```

Comment:
Two things about line 107.  
1.  Avoid putting litteral strings in your code.  Instead put a const 'PhysicianURL' at the top of the class and Assert with PhysicianURL.
2. You will have different enviroments that you tests will run against, such as your own dev machine, QA, Staging, and even production.  The url will be different for each (eg. staging.iemosoft.com, qa.iemosoft.com,).  Often I'll do a contains (eg. var PhysURLSnippet = "Seleneium-test-physician";  Assert.IsTrue(doc_url.contains(PhysURLSnippet))

Reply:
yeah, the constant and sub-string comparison defiantly makes sense.